instrumentation/spring/spring-kafka-2.7/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/kafka/AbstractMessageListenerContainerInstrumentation.java

```diff
@@ -56,13 +53,13 @@ public static class GetBatchInterceptorAdvice {
     public static <K, V> void onExit(
         @Advice.Return(readOnly = false) BatchInterceptor<K, V> interceptor) {
 
-      if (!(interceptor instanceof InstrumentedBatchInterceptor)) {
-        VirtualField<ConsumerRecords<K, V>, Context> receiveContextField =
-            VirtualField.find(ConsumerRecords.class, Context.class);
-        VirtualField<ConsumerRecords<K, V>, State<ConsumerRecords<K, V>>> stateField =
-            VirtualField.find(ConsumerRecords.class, State.class);
-        interceptor =
-            new InstrumentedBatchInterceptor<>(receiveContextField, stateField, interceptor);
+      if (interceptor == null
+          || !interceptor
+              .getClass()
+              .getName()
+              .equals(
+                  "io.opentelemetry.instrumentation.spring.kafka.v2_7.InstrumentedBatchInterceptor")) {
```

Comment:
what would break if you'd use `InstrumentedBatchInterceptor.class.getName()` or `interceptor.getClass() != InstrumentedBatchInterceptor.class` here? Did you choose to use class name here to keep `InstrumentedBatchInterceptor` package private?

Reply:
If it is do keep the class package private WDYT about using something like `telemetry().isInstrumentedBatchInterceptor(interceptor)`